### ISSUE TITLE
Checkout Address Component

### ISSUE THREAD

**Issue by spencern** [reactioncommerce/reaction-component-library#111]
# Checkout Shipping Address

### Summary description of UI component
This is a component that permits the customer to add a shipping address to their checkout. The shipping address they add may be entered via an address form or by selecting from a list of previously saved addresses.
If an address has been chosen, this component is responsible for displaying the chosen address and providing an action for which the user can choose or enter a different address.

### Rationale for why this component is necessary
This component is needed to wrap address book components and render the correct address component based on the data in the cart.

### Expected use cases
This component will be used in the checkout process

### Where this component should be created
This component should be created in the component library

### Images / Designs of UI component in context
Enter/Add Address
<img width="639" alt="image" src="https://user-images.githubusercontent.com/1203639/42172541-86c5f522-7dd9-11e8-989d-8177e35a3530.png">

Resolve Address Validation Errors
![image](https://user-images.githubusercontent.com/1203639/42172524-786d3ddc-7dd9-11e8-8f44-7ac92baff2fa.png)

Choose Address
![image](https://user-images.githubusercontent.com/1203639/42173504-7a7fb278-7ddc-11e8-9925-853a8d2814bb.png)

Edit Address
![image](https://user-images.githubusercontent.com/1203639/42173652-02ee921e-7ddd-11e8-9d9a-02cdc3e2d5fc.png)

```jsx
// INCOMPLETE Address Action
// Pseudo-markup for the Checkout Shipping Address Action
<div>
  <header>{{StepNumber}}. Enter|Choose|Add Shipping Address</header>
  <AddressBookComponent />
</div>
```

```jsx
// COMPLETE Address Action
// Pseudo-markup for the Checkout Shipping Address Action
<div>
  <CheckoutActionComplete
    label="Shipping Address"
    content="addressData"
    change="handleAddressChange"
  />
</div>
```

### State
This component is either `complete` or `incomplete`
```js
// If the customer is signed in and has an existing address in their address book,
// the status should default to complete, and the default address should be used by default.
state: {
  status: "incomplete"
}
```

### Props
```js
propTypes: {
  isLoading: PropTypes.bool,
  shippingAddress: PropTypes.shape({
    id: PropTypes.string,
    country: PropTypes.string,
    fullName: PropTypes.string,
    address1: PropTypes.string,
    address2: PropTypes.string,
    region: PropTypes.string,
    postal: PropTypes.string,
    phone: PropTypes.string
  }),
}
```

### Render Criteria
This component will be rendered whenever a shipping address is needed in the checkout process.

#### Normal
The Normal and Empty states are identical in this component
In the normal, incomplete state, there is no shipping address on the cart. The component should render the header and the new address form.

#### Empty
When the shipping address on the cart is missing, the component should render the address header and the new address form.

#### Complete
When the shipping address exists on the cart, the component should render the `CheckoutStepComplete` component with the address on the cart.

#### Loading / retrieving data
When the component is waiting on data (such as from the address service) the component should render the header and a spinner.

#### Processing / Waiting for response
This component shouldn't have a processing state, though child components such as the address form will.

#### Error
This component should not have an error state

**Comment by spencern** [comment]
Closing this in favor of #115
